README.md

```diff
@@ -7,19 +7,29 @@ Overview
 Laser scan processing tools. The meta-package contains:
 
  * `laser_ortho_projector`: calculates orthogonal projections of LaserScan messages
+    - License: BSD-3-Clause
  
  * `laser_scan_matcher`: an incremental laser scan matcher, using Andrea Censi's Canonical 
-Scan Matcher implementation. It downloads and installs Andrea Censi's Canonical Scan Matcher [1] locally.
+Scan Matcher (CSM) implementation [1].
+    - License: BSD-3-Clause
+    - Note: CSM is LGPL-3.0 licensed and depends on either [GSL](https://www.gnu.org/software/gsl/), which is GPL-3.0 or [eigen](https://eigen.tuxfamily.org/), which is MPL-2.0, depending on the version of CSM.
```

Comment:
> CSM is LGPL-3.0 licensed and depends on either [GSL](https://www.gnu.org/software/gsl/), which is GPL-3.0 or (...)

Is it LGPL with GPL present in the dependency chain? If https://github.com/CCNYRoboticsLab/scan_tools/issues/76#issuecomment-1185830184 is correct, then would it need to be corrected to `GPL` (unfortunately)?

I'm not so sure. But if that's right we might also have to correct `package.xml`.

Reply:
I went through all of the source files of each package to see what the license was declared at the source file level.   For laser_scan_matcher it was all BSD.

Also, the original manifest file specifies BSD: https://github.com/CCNYRoboticsLab/scan_tools/blob/8318391ae0fbcfb26720ba8c6dd4a49ce8a1f0fc/laser_scan_matcher/manifest.xml

When it was converted to catkin the LGPL line was added to the package.xml file:
https://github.com/CCNYRoboticsLab/scan_tools/commit/8318391ae0fbcfb26720ba8c6dd4a49ce8a1f0fc

It's ok to be dual licensed, but it's possible LGPL was added by mistake.  I didn't remove it from the package.xml, but it's probably safest to also reference LGPL here in the readme.

> Is it LGPL with GPL present in the dependency chain? If https://github.com/CCNYRoboticsLab/scan_tools/issues/76#issuecomment-1185830184 is correct, then would it need to be corrected to GPL (unfortunately)?

Having a dependency be GPL licensed does not cause this package to also be GPL licensed.  Both LGPL and BSD are compatible with GPL.  It DOES require people who use that GPL library (transitively, through laser_scan_matcher) to also follow the copyleft requirements of GPL imposed by that dependency.

This is problematic, practically, and should be called out in the readme, which the MR tries to do by notifying the downstream user what the licensing implications of the CSM dependency are depending on the version used.

Reply:
Ok, I updated the license line to include both BSD and LGPL to be consistent with the package.xml file